### PR TITLE
Pass sys.executable in debugger config

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Depends on https://github.com/jupyter-xeus/xeus/pull/259.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
@JohanMabille do you think we should merge as-is?

File: src/xdebugger.cpp [hunk 1]
```diff
@@ -128,8 +130,16 @@ namespace xpyt
...
     {
-        // debugpy has to be started in the main thread
-        std::string code = "import debugpy\ndebugpy.listen((\'" + m_debugpy_host + "\'," + m_debugpy_port + "))";
+        // import debugpy 
+        std::string code = "import debugpy;";
+        // specify sys.executable
+        auto it = m_debugger_config.find("python");
+        if (it != m_debugger_config.end())
+        {
+            code += "debugpy.configure({\'python\': \'" + it->template get<std::string>()  + "\'});";
```
Comment (JohanMabille) [inline src/xdebugger.cpp:139]: I think you can remove the `;` before the last `"`

File: src/main.cpp [hunk 2]
```diff
@@ -102,14 +102,11 @@ int main(int argc, char* argv[])
...
-// On windows, sys.executable is not properly set with Py_SetProgramName
-// Cf. https://bugs.python.org/issue34725
-// A private undocumented API was added as a workaround in Python 3.7.2.
-#ifdef _WIN32
-    _Py_SetProgramFullPath(const_cast<wchar_t*>(wexecutable.c_str()));
-#else
+    // On windows, sys.executable is not properly set with Py_SetProgramName
+    // Cf. https://bugs.python.org/issue34725
+    // A private undocumented API was added as a workaround in Python 3.7.2.
+    // _Py_SetProgramFullPath(const_cast<wchar_t*>(wexecutable.c_str()));
```
Comment (JohanMabille) [inline src/main.cpp:108]: Should we call this private API when Python's version is greater than 3.7.1?

Comment (SylvainCorlay) [reply src/main.cpp:108]: We could do it separately.

File: src/xdebugger.cpp (hunk 1 above)
Comment (SylvainCorlay) [reply src/xdebugger.cpp:139]: Don't we do `code += "debugpy.listen` right after?

Comment (JohanMabille) [reply src/xdebugger.cpp:139]: Ha my bad, I was persuaded the statements were separated with ' \n'.

File: src/main.cpp (hunk 2 above)
Comment (JohanMabille) [reply src/main.cpp:108]: Ok!

**Pull request merged**